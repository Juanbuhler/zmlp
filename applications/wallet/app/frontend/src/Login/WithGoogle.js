import { useState, useEffect } from 'react'
import getConfig from 'next/config'
import Head from 'next/head'

import Button, { VARIANTS } from '../Button'

const {
  publicRuntimeConfig: { GOOGLE_OAUTH_API_KEY },
} = getConfig()

let GoogleAuth

const LoginWithGoogle = () => {
  const [isLoggedIn, setIsLoggedIn] = useState(false)

  useEffect(() => {
    window.onload = () => {
      window.gapi.load('auth2', async () => {
        GoogleAuth = await window.gapi.auth2.init({
          client_id: `${GOOGLE_OAUTH_API_KEY}.apps.googleusercontent.com`,
        })
        setIsLoggedIn(GoogleAuth.isSignedIn.get())
      })
    }
  })

  return (
    <>
      <Head>
        <script src="https://apis.google.com/js/platform.js" async defer />
      </Head>

      <Button
        variant={VARIANTS.PRIMARY}
        isDisabled={false}
        onClick={async () => {
          if (isLoggedIn) {
            await GoogleAuth.signOut()
            setIsLoggedIn(false)
          } else {
            const response = await GoogleAuth.signIn()
            setIsLoggedIn(true)
            const idToken = response.Zi.id_token

            console.warn(idToken)

            await fetch('/api/v1/projects/', {
              headers: {
                'content-type': 'application/json;charset=UTF-8',
                Authorization: `Bearer ${idToken}`,
              },
            })
          }
        }}>
        {isLoggedIn ? 'Sign Out' : 'Sign In with Google'}
      </Button>
    </>
  )
}

export default LoginWithGoogle
