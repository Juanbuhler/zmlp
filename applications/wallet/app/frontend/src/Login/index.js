import { useState } from 'react'
import PropTypes from 'prop-types'

import { colors, constants, typography, spacing } from '../Styles'

import LogoSvg from '../Icons/logo.svg'
import HiddenSvg from '../Icons/hidden.svg'
import VisibleSvg from '../Icons/visible.svg'

import FormAlert from '../FormAlert'
import Input from '../Input'
import Button, { VARIANTS } from '../Button'

const WIDTH = 440
const LOGO_WIDTH = 143

const Login = ({ errorMessage, setErrorMessage, onSubmit }) => {
  const [username, setUsername] = useState('')
  const [password, setPassword] = useState('')
  const [showPassword, setShowPassword] = useState(false)

  return (
    <div
      css={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        height: '100vh',
      }}>
      <form
        method="post"
        onSubmit={event => event.preventDefault()}
        css={{
          display: 'flex',
          flexDirection: 'column',
          padding: `${spacing.normal}px ${spacing.colossal}px`,
          width: WIDTH,
          backgroundColor: colors.secondaryBackground,
          borderRadius: constants.borderRadius.small,
          boxShadow: constants.boxShadows.default,
        }}>
        <LogoSvg
          width={LOGO_WIDTH}
          css={{
            alignSelf: 'center',
            paddingTop: spacing.comfy,
            paddingBottom: spacing.comfy,
          }}
        />
        <h3
          css={{
            textAlign: 'center',
            fontSize: typography.size.mega,
            lineHeight: typography.height.mega,
            margin: 0,
            paddingTop: spacing.moderate,
            paddingBottom: spacing.moderate,
          }}>
          Welcome. Please login.
        </h3>
        <FormAlert
          errorMessage={errorMessage}
          setErrorMessage={setErrorMessage}
        />
        <Input
          autoFocus
          id="username"
          label="Email"
          type="text"
          value={username}
          onChange={({ target: { value } }) => setUsername(value)}
          hasError={!!errorMessage}
        />
        <Input
          id="password"
          label="Password"
          type={showPassword ? 'text' : 'password'}
          value={password}
          onChange={({ target: { value } }) => setPassword(value)}
          hasError={!!errorMessage}
          after={
            <Button
              aria-label={showPassword ? 'Hide password' : 'Show password'}
              variant={VARIANTS.NEUTRAL}
              onClick={() => setShowPassword(!showPassword)}
              style={{ '&:hover': { color: colors.primary } }}>
              {showPassword ? (
                <VisibleSvg width={20} />
              ) : (
                <HiddenSvg width={20} />
              )}
            </Button>
          }
        />
        <div
          css={{
            padding: spacing.comfy,
            display: 'flex',
            justifyContent: 'center',
          }}>
          <Button
            type="submit"
            variant={VARIANTS.PRIMARY}
            onClick={() => onSubmit({ username, password })}
            isDisabled={!username || !password}>
            Login
          </Button>
        </div>
      </form>
    </div>
  )
}

Login.propTypes = {
  errorMessage: PropTypes.string.isRequired,
  setErrorMessage: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
}

export default Login
