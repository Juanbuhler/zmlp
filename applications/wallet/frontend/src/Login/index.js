import { useState } from 'react'
import PropTypes from 'prop-types'
import Head from 'next/head'
import Link from 'next/link'
import { useRouter } from 'next/router'

import { colors, constants, typography, spacing } from '../Styles'

import LargeLogo from '../Icons/largeLogo.svg'
import HiddenSvg from '../Icons/hidden.svg'
import VisibleSvg from '../Icons/visible.svg'

import FlashMessage, { VARIANTS as FLASH_VARIANTS } from '../FlashMessage'
import Input, { VARIANTS as INPUT_VARIANTS } from '../Input'
import Button, { VARIANTS as BUTTON_VARIANTS } from '../Button'

import LoginWithGoogle from './WithGoogle'

const WIDTH = 446
const LOGO_WIDTH = 180

const Login = ({ googleAuth, hasGoogleLoaded, errorMessage, onSubmit }) => {
  const {
    query: { action },
  } = useRouter()

  const [username, setUsername] = useState('')
  const [password, setPassword] = useState('')
  const [showPassword, setShowPassword] = useState(false)

  return (
    <div
      css={{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        height: '100vh',
      }}
    >
      <Head>
        <title>Login</title>
      </Head>

      {action === 'password-reset-request-success' && (
        <div css={{ paddingBottom: spacing.normal }}>
          <FlashMessage variant={FLASH_VARIANTS.SUCCESS}>
            Password reset email sent.
          </FlashMessage>
        </div>
      )}

      {action === 'password-reset-update-success' && (
        <div css={{ paddingBottom: spacing.normal }}>
          <FlashMessage variant={FLASH_VARIANTS.SUCCESS}>
            Password has been updated.
          </FlashMessage>
        </div>
      )}

      {action === 'account-activation-success' && (
        <div css={{ paddingBottom: spacing.normal }}>
          <FlashMessage variant={FLASH_VARIANTS.SUCCESS}>
            Account activated. Please login now.
          </FlashMessage>
        </div>
      )}

      <form
        method="post"
        onSubmit={(event) => event.preventDefault()}
        css={{
          display: 'flex',
          flexDirection: 'column',
          padding: spacing.colossal,
          width: WIDTH,
          backgroundColor: colors.structure.mattGrey,
          borderRadius: constants.borderRadius.small,
          boxShadow: constants.boxShadows.default,
        }}
      >
        <LargeLogo width={LOGO_WIDTH} css={{ alignSelf: 'center' }} />

        <h3
          css={{
            textAlign: 'center',
            fontSize: typography.size.large,
            lineHeight: typography.height.large,
            paddingTop: spacing.spacious,
            paddingBottom: spacing.spacious,
          }}
        >
          Welcome. Please login.
        </h3>

        <LoginWithGoogle
          googleAuth={googleAuth}
          hasGoogleLoaded={hasGoogleLoaded}
          onSubmit={onSubmit}
        />

        {!!errorMessage && (
          <div
            css={{
              paddingBottom: spacing.base,
            }}
          >
            <FlashMessage variant={FLASH_VARIANTS.ERROR}>
              {errorMessage}
              <br />
              <a
                href="mailto:support@zorroa.com?subject=ZVI Console Login Trouble"
                target="_blank"
                rel="noopener noreferrer"
              >
                Contact Support
              </a>
            </FlashMessage>
          </div>
        )}

        <Input
          autoFocus
          id="username"
          variant={INPUT_VARIANTS.PRIMARY}
          label="Email"
          type="text"
          value={username}
          onChange={({ target: { value } }) => setUsername(value)}
          hasError={!!errorMessage}
        />

        <Input
          id="password"
          variant={INPUT_VARIANTS.PRIMARY}
          label="Password"
          type={showPassword ? 'text' : 'password'}
          value={password}
          onChange={({ target: { value } }) => setPassword(value)}
          hasError={!!errorMessage}
          after={
            <Button
              aria-label={showPassword ? 'Hide password' : 'Show password'}
              variant={BUTTON_VARIANTS.NEUTRAL}
              onClick={() => setShowPassword(!showPassword)}
              style={{
                color: colors.structure.zinc,
                padding: spacing.moderate,
                outlineOffset: -2,
                '&:hover': { color: colors.key.one },
              }}
            >
              {showPassword ? (
                <VisibleSvg height={constants.icons.regular} />
              ) : (
                <HiddenSvg height={constants.icons.regular} />
              )}
            </Button>
          }
        />

        <div
          css={{
            paddingTop: spacing.normal,
            display: 'flex',
            justifyContent: 'center',
          }}
        >
          <Button
            type="submit"
            variant={BUTTON_VARIANTS.PRIMARY}
            onClick={() => onSubmit({ username, password })}
            isDisabled={!username || !password}
          >
            Login
          </Button>
        </div>

        <div
          css={{
            paddingTop: spacing.spacious,
            textAlign: 'center',
            a: {
              color: colors.structure.steel,
            },
          }}
        >
          <Link href="/create-account">
            <a>Create an account</a>
          </Link>
          <br />
          <br />
          <Link href="/reset-password">
            <a>Reset your password</a>
          </Link>
        </div>
      </form>
    </div>
  )
}

Login.propTypes = {
  googleAuth: PropTypes.shape({
    signIn: PropTypes.func.isRequired,
  }).isRequired,
  hasGoogleLoaded: PropTypes.bool.isRequired,
  errorMessage: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
}

export default Login
