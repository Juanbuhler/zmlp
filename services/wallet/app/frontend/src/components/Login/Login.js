import React, { useState, useRef } from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import { Redirect, Link } from 'react-router-dom'

import User from '../../models/User'
import Page from '../Page'
import Logo from '../Logo'

const ERROR_MESSAGE = 'Invalid email or password'

function Login({ user, login, history }) {
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const [error, setError] = useState('')
  const emailInput = useRef(null)

  function handleSubmit(e) {
    e.preventDefault()
    setError('')

    if (email !== '' && password !== '') {
      login(email, password)
        .then(() => {
          history.push('/workspace')
        })
        .catch(() => {
          setError(ERROR_MESSAGE)
          emailInput.current.focus()
        })
    } else {
      setError(ERROR_MESSAGE)
      emailInput.current.focus()
    }
  }

  if (user.attrs.tokens) {
    return <Redirect to={'/workspace'} />
  }

  return (
    <Page>
      <div className="login__page">
        <form className="login__form" onSubmit={handleSubmit}>
          <Logo width="143" height="42" />
          <h3 className="login__form-heading">Welcome. Please login.</h3>

          {/*
          (loading) && (
            <p className="login__form-loader">Loading</p>
          )
          */}

          {error && (
            <div className="login__form-error-container">
              <i className="fas fa-exclamation-triangle"></i>
              <p className="login__form-error-message">{error}</p>
            </div>
          )}

          {/* <p className="login__form-sub-heading">- or -</p> */}

          <div className="login__form-group">
            <label className="login__form-label" htmlFor="email">
              Email
            </label>
            <input
              id="email"
              ref={emailInput}
              type="text"
              value={email}
              name="email"
              className={cx(
                error ? 'login__form-input--error' : 'login__form-input',
              )}
              onChange={e => {
                setEmail(e.target.value)
                setError('')
              }}
            />
          </div>

          <div className="login__form-group">
            <label className="login__form-label" htmlFor="password">
              Password
            </label>
            <input
              id="password"
              type="password"
              value={password}
              name="password"
              className={cx(
                error ? 'login__form-input--error' : 'login__form-input',
              )}
              onChange={e => {
                setPassword(e.target.value)
                setError('')
              }}
            />
          </div>

          <button
            className="login__btn btn btn-primary"
            type="submit"
            disabled={error.length}>
            Login
          </button>

          <Link to="/" className="login__form-tagline">
            Forgot Password? Need login help?
          </Link>
        </form>
      </div>
    </Page>
  )
}

Login.propTypes = {
  user: PropTypes.instanceOf(User).isRequired,
  login: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired,
}

export default Login
