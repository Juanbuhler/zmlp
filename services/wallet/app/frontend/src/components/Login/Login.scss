@import '../../styles/core.scss';

.login__page {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.login__form {
  display: flex;
  flex-direction: column;
  padding: 65px;
  width: 440px;
  height: 580px;
  // background-color: lighten($background-color, 25%);
  background-color: #2d2d2d;
  border-radius: 2px;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.59);
}

.login__form-heading {
  text-align: center;
}

.login__form-group {
  margin-bottom: 15px;
}

.login__form-input {
  width: 100%;
}

.login__btn {
  width: 95px;
  margin: 30px auto;
  font-weight: $bold-font-weight;
}

.login__form-tagline {
  text-align: center;
  opacity: 0.54;
}
