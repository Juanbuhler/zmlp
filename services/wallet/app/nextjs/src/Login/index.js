import { useState } from 'react'
import PropTypes from 'prop-types'

import { colors, constants, typography, spacing } from '../Styles'

import Input from '../Input'

import LogoSvg from './logo.svg'

const WIDTH = 440
const HEIGHT = 580
const LOGO_WIDTH = 143

const Login = ({ onSubmit }) => {
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')

  return (
    <div
      css={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        height: '100%',
      }}>
      <form
        css={{
          display: 'flex',
          flexDirection: 'column',
          justifyContent: 'space-evenly',
          padding: `${spacing.spacious}px ${spacing.colossal}px`,
          width: WIDTH,
          height: HEIGHT,
          backgroundColor: colors.secondaryBackground,
          borderRadius: constants.borderRadius.small,
          boxShadow: constants.boxShadows.default,
        }}>
        <LogoSvg width={LOGO_WIDTH} css={{ margin: '0 auto' }} />
        <h3
          css={{
            textAlign: 'center',
            fontWeight: typography.weight.medium,
            paddingBottom: spacing.spacious,
          }}>
          Welcome. Please login.
        </h3>
        <Input
          id="email"
          label="Email"
          type="text"
          value={email}
          onChange={({ target: { value } }) => setEmail(value)}
        />
        <Input
          id="password"
          label="Password"
          type="password"
          value={password}
          onChange={({ target: { value } }) => setPassword(value)}
        />
        <div css={{ padding: spacing.spacious, textAlign: 'center' }}>
          <button
            type="button"
            onClick={onSubmit}
            css={{
              backgroundColor: colors.primary,
              color: colors.primaryFont,
              width: '95px',
              fontSize: typography.size.hecto,
              lineHeight: typography.height.hecto,
              fontWeight: typography.weight.medium,
              borderRadius: constants.borderRadius.small,
              padding: `${spacing.moderate}px ${spacing.spacious}px`,
              border: 0,
              cursor: 'pointer',
              ':hover': {
                backgroundColor: colors.primaryHover,
              },
            }}>
            Login
          </button>
        </div>
        <a href="/" css={{ textAlign: 'center' }}>
          Forgot Password? Need login help?
        </a>
      </form>
    </div>
  )
}

Login.propTypes = {
  onSubmit: PropTypes.func.isRequired,
}

export default Login
