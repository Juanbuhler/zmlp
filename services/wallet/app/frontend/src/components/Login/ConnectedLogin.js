import React from 'react'

import { useAuth } from '../../context/authContext'
import { useUser } from '../../context/userContext'
import Login from './Login'

function ConnectedLogin(props) {
  const authContext = useAuth()
  const userContext = useUser()

  return <Login {...authContext} {...userContext} {...props} />
}
export default ConnectedLogin
