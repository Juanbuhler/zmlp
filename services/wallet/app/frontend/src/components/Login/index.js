import Login from './ConnectedLogin'
export default Login
