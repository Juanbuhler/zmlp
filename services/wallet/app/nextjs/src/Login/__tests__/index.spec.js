import TestRenderer from 'react-test-renderer'

import Login from '../'

const noop = () => () => {}

describe('<Login />', () => {
  it('should render properly', () => {
    const component = TestRenderer.create(<Login onSubmit={noop} />)

    expect(component.toJSON()).toMatchSnapshot()
  })
})
