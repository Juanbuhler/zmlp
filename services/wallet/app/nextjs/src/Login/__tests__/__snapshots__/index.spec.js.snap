// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render properly 1`] = `
.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  padding: 32px 64px;
  width: 440px;
  height: 580px;
  background-color: #2d2d2d;
  border-radius: 2px;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.59);
}

.emotion-0 {
  margin: 0 auto;
}

.emotion-1 {
  text-align: center;
  font-weight: 500;
  padding-bottom: 32px;
}

.emotion-2 {
  display: block;
  padding-bottom: 12px;
}

.emotion-3 {
  height: 39px;
  border-radius: 2px;
  box-shadow: inset 0 1px 3px 0 transparent;
  width: 100%;
}

.emotion-7 {
  padding: 32px;
  text-align: center;
}

.emotion-6 {
  background-color: #82be3c;
  color: #fff;
  width: 95px;
  font-size: 14px;
  line-height: 16px;
  font-weight: 500;
  border-radius: 2px;
  padding: 12px 32px;
  border: 0;
  cursor: pointer;
}

.emotion-6:hover {
  background-color: #a7d374;
}

.emotion-8 {
  text-align: center;
}

<div
  className="emotion-10"
>
  <form
    className="emotion-9"
  >
    <svg
      className="emotion-0"
      data-file-name="logo.svg"
      width={143}
    />
    <h3
      className="emotion-1"
    >
      Welcome. Please login.
    </h3>
    <div>
      <label
        className="emotion-2"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="emotion-3"
        id="email"
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="emotion-2"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="emotion-3"
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="emotion-7"
    >
      <button
        className="emotion-6"
        type="button"
      >
        Login
      </button>
    </div>
    <a
      className="emotion-8"
      href="/"
    >
      Forgot Password? Need login help?
    </a>
  </form>
</div>
`;
